articles/quickstart/native/ionic4/01-login.md

```diff
@@ -0,0 +1,405 @@
+---
+title: Login
+default: true
+description: This tutorial demonstrates how to add user login to an Ionic 4 application using Auth0.
+budicon: 448
+topics:
+  - quickstarts
+  - native
+  - ionic4
+github:
+  path: 01-Login
+contentType: tutorial
+useCase: quickstart
+---
+
+<%= include('../_includes/_getting_started', { library: 'Ionic' }) %>
+
+<%= include('../../../_includes/_callback_url') %>
+
+The **Callback URL** to be used for your application includes your app's package ID which is found in the `config.xml` file for your app.
+
+Go to the <a href="${manage_url}/#/applications/${account.clientId}/settings">Application Settings</a> section in your Auth0 dashboard and set your **Callback URL** in the **Allowed Callback URLs** box.
+
+If you are following along with the sample project you downloaded from the top of this page, you should set the **Allowed Callback URL** to
+
+```bash
+# replace YOUR_PACKAGE_ID with your app package ID
+YOUR_PACKAGE_ID://${account.namespace}/cordova/YOUR_PACKAGE_ID/callback
+```
+
+Replace `YOUR_PACKAGE_ID` with your application's package name, available as the `applicationId` attribute in the `app/build.gradle` file.
+
+
+<%= include('../../../_includes/_logout_url') %>
+
+::: note
+If you are following along with the sample project you downloaded from the top of this page, the logout URL you need to whitelist in the Allowed Logout URLs field is the same as the callback URL.
+:::
+
+Add `file` as an allowed origin to the **Allowed Origins (CORS)** box.
+
+```bash
+file://*
+```
+
+Lastly, be sure that the **Application Type** for your application is set to **Native** in the application settings.
+
+## Install the Dependencies
+
+The required dependencies for using Auth0 in an Ionic application are **auth0.js** and **auth0-cordova**. Install them with npm or yarn.
+
+```bash
+# installation with npm
+npm install auth0-js @auth0/cordova --save
+
+# installation with yarn
+yarn add auth0-js @auth0/cordova
+```
+
+Since `angular-cli` 6 and up pollyfills for node core libraries are disabled by default. As auth0-cordova uses `crypto` you'll need to provide a pollyfill for this library.
```

Comment:
I'm thinking maybe rewording this to

```
Auth0-Cordova requires the node library `crypto`, however since Angular CLI version 6+ you are now required to use a polyfill for this library.  You can accomplish this using a custom webpack configuration.
```

thoughts?